lib/Frontend/CompilerInvocation.cpp

```diff
@@ -634,6 +634,9 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
     // If this is a known experimental feature, allow it in +Asserts
     // (non-release) builds for testing purposes.
     if (auto feature = getExperimentalFeature(A->getValue())) {
+      if (isFeatureAvailableInProduction(*feature))
+        continue;
```

Comment:
You'll need to `Opts.Features.insert(*feature);` here.

Reply:
Oops, fixed, thank you!